src/sentry/search/events/builder.py

```diff
@@ -1645,12 +1660,133 @@ def __init__(
 
 
 class SessionsQueryBuilder(QueryBuilder):
+    # ToDo(ahmed): Rename this to AlertsSessionsQueryBuilder as it is exclusively used for crash rate alerts
     def resolve_params(self) -> List[WhereType]:
         conditions = super().resolve_params()
         conditions.append(Condition(self.column("org_id"), Op.EQ, self.organization_id))
         return conditions
 
 
+class SessionsV2QueryBuilder(QueryBuilder):
+    condition_fields = {"project", "project_id", "environment", "release"}
+
+    def __init__(
+        self,
+        *args: Any,
+        granularity: Optional[int] = None,
+        extra_condition_fields: Optional[Sequence[str]] = None,
+        **kwargs: Any,
+    ):
+        self._granularity = granularity
```

Comment:
Could we just resolve it instantly instead?
```suggestion
        self.granularity = Granularity(granularity) if granularity is not None else None
```